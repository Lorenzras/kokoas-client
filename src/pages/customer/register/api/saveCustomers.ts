import { APPIDS, KintoneRecord } from '../../../../api/kintone';
import { CustomerForm } from '../form';
import { formToKintCust } from '../helper';

export const saveCustomers = async (formData: CustomerForm) => {
  const transformedCust = formToKintCust(formData);

  /* Create record */
  if (!formData.id){
    return KintoneRecord.addRecords({
      app: APPIDS.customers,
      records: transformedCust,
    }).then(resp => resp.records);
  }


  /* Save new customer in edit mode */
  const unsavedCust = transformedCust.filter(cust => !cust.$id?.value);
  let savedRecords = [] as { id: string, revision: string }[];

  if (unsavedCust.length > 0){
    savedRecords = await KintoneRecord.addRecords({
      app: APPIDS.customers,
      records: unsavedCust,
    }).then(resp => resp.records);
  }

  /* Update record */
  return KintoneRecord.updateRecords({
    app: APPIDS.customers,
    records: transformedCust
      .filter(cust => !!cust.$id?.value)
      .map(cust => {
        return  {
          id: cust.$id!.value,
          record: cust,
        };
      }),
  }).then(resp => resp.records.concat(savedRecords));


};


