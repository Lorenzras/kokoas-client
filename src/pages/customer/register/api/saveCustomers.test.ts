import { saveCustomers } from './saveCustomers';
import { testData } from './test/testData';


test('Save', async ()=>{
  expect(await saveCustomers(testData)).toMatchSnapshot();
});