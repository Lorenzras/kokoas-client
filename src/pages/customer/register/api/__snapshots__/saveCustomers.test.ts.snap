// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Save 1`] = `
Array [
  Object {
    "id": "368",
    "revision": "1",
  },
  Object {
    "id": "369",
    "revision": "1",
  },
]
`;
